libopenage/renderer/stages/world/render_stage.cpp

```diff
@@ -156,4 +181,37 @@ void WorldRenderStage::init_uniform_ids() {
 	WorldObject::anchor_offset = this->display_shader->get_uniform_id("anchor_offset");
 }
 
+void WorldRenderStage::initialize_render_pass_with_shader_commands(size_t width, size_t height, const util::Path &shaderdir, const util::Path &config_path) {
+	auto vert_shader_file = (shaderdir / "demo_7_world.vert.glsl").open();
+	auto vert_shader_src = renderer::resources::ShaderSource(
+		resources::shader_lang_t::glsl,
+		resources::shader_stage_t::vertex,
+		vert_shader_file.read());
+	vert_shader_file.close();
+
+	auto frag_shader_file = (shaderdir / "demo_7_world.frag.glsl").open();
+	log::log(INFO << "Loading shader commands config from: " << (shaderdir / "demo_7_display.frag.glsl"));
+	this->shader_template = std::make_shared<ShaderCommandTemplate>(frag_shader_file.read());
+	if (not this->shader_template->load_commands(config_path / "world_commands.config")) {
+		log::log(ERR << "Failed to load shader commands configuration for world stage");
+		return;
```

Comment:
You should raise an Exception in this case because the stage cannot be initialized properly. It's in an unrecoverable state if the shader cannot be loaded.